doc/api/process.md

```diff
@@ -1917,6 +1917,44 @@ console.log('After:', getActiveResourcesInfo());
 //   After: [ 'TTYWrap', 'TTYWrap', 'TTYWrap', 'Timeout' ]
 ```
 
+## `process.getBuiltin(id)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `id` {string} ID of the built-in module being requested. Must not contain the `node:` prefix.
+* Returns: {Object}
+
+`process.getBuiltin(id)` provides a way to load the built-in modules
+in a globally available function. ES Modules that need to support
+other environments can use it to conditionally load a Node.js builtin
+when it is run in Node.js, without having to deal with the resolution
+error that can be thrown by `import 'node:id'` in a non-Node.js
+environment or having to use dynamic `import()` which either turns the
+module into an asynchronous module, or turns a synchronous API into an
+asynchronous one.
+
+```mjs
+if (globalThis.process && globalThis.process.getBuiltin) {
+  // Run in Node.js, use the Node.js fs module.
+  const fs = globalThis.process.getBuiltin('fs');
+  // If `require()` is needed to load user-modules, use createRequire()
+  const m = globalThis.process.getBuiltin('module');
+  const require = m.createRequire(import.meta.url);
+  const foo = require('foo');
+}
+```
+
+If `id` specifies a built-in available in the current Node.js process,
+`process.getBuiltin()` method returns the corresponding built-in
+module, which is the same as the object returned by
+[`require(id)`][`require()`]. If `id` does not correspond to any
+built-in module, an [`ERR_UNKNOWN_BUILTIN_MODULE`][] would be thrown.
```

Comment:
```suggestion
built-in module, an [`ERR_UNKNOWN_BUILTIN_MODULE`][] is thrown.
```

Reply:
I would actually prefer that the method not throw and return `undefined` in this case. I'm imagining a use case where another runtime can implement this method and it can be used to determine if that runtime actually provides an implementation of that built-in. For instance, in workers, we provide `node:stream` but not `node:fs` ... in the latter case, `process.getBuiltin(...)` returning `undefined` is a bit more optimal than creating an error that I'm likely just going to throw away and handle with a fallback.

Reply:
Returning a falsy value SGTM, although I think the last time I looked into it I recall seeing an unrelated API returning null (unfortunately I cannot remember what API this is now...). I wonder if null would be better than undefined as that is a more consistent typeof.

Reply:
I'd personally prefer undefined over null... I'm not sure what you mean by "a more consistent typeof" when `typeof null === "object"`.

Reply:
That being said, I'm not sure why thowing is a big deal when that's already what would happen if you fail a require today...

Reply:
undefined would be what my instincts say too but consistency seems important (but then, I cannot remember what the precedent is, maybe it was only something I saw in my dream somehow?)

Reply:
Updated to return undefined for unfound modules.